controllers/hostdefinition_controller.go

```diff
@@ -0,0 +1,456 @@
+/**
```

Comment:
look at the description, I opened a ticket for it because I didn't want that the diff will be bigger

Reply:
I think that including CSI-4185 here would actually reduce the diff

Reply:
it will change in places in the controller of the csi that I didn't edit in this PR, so I think it will increase the diff but if you think I should do it I will do it

Reply:
I think it would be better, yes

Reply:
another option is to make the code more generic that way in a separate PR, and merge that separate PR before this one

Reply:
https://github.com/IBM/ibm-block-csi-operator/pull/258 PR solves it